### PR TITLE
Derive smoothness from `mtb:scale` and `tracktype`

### PR DESCRIPTION
**Pull request by rush42**
This PR implements a mapping from the tags 'mtb:scale` and `tracktype` to derive values for smoothness.
Closes [973](https://github.com/FixMyBerlin/private-issues/issues/973).
The mapping is based on [Wiki tracktype ](https://wiki.openstreetmap.org/wiki/Key:tracktype) and [Wiki mtb:scale](https://wiki.openstreetmap.org/wiki/Key:mtb:scale#mtb:scale=0-6)

### CODE REVIEW

**Comment by rush42** [comment]
@tordans please check if the mapping I suggested works out for you

File: app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua [hunk 1]
```diff
@@ -99,3 +99,31 @@ function SmoothnessFromSurface(surface)
   end
   return smoothness, source, confidence, todo
 end
+
+function SmoothnessFromMTBScale(scale)
```
Comment (tordans) [inline app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua:103]:
```suggestion
-- https://wiki.openstreetmap.org/wiki/Key:mtb:scale
-- https://taginfo.openstreetmap.org/keys/mtb%3Ascale#values
function SmoothnessFromMTBScale(scale)
```

File: app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua [hunk 2]
```diff
@@ -99,3 +99,31 @@ function SmoothnessFromSurface(surface)
...
+  if not type then
+    return nil, nil, nil, "No track type was given"
+  end
+  local trackTypeToSmoothness = {
+    ["grade1"] = "good",
+    ["grade2"] = "intermediate",
+    ["grade3"] = "intermediate",
+    ["grade4"] = "bad",
+    ["grade5"] = "very_bad"
+  }
```
Comment (tordans) [inline app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua:123]:
```suggestion
  local trackTypeToSmoothness = {
    ["grade1"] = "good",
    ["grade2"] = "intermediate",
    ["grade3"] = "bad",
    ["grade4"] = "bad",
    ["grade5"] = "very_bad"
  }
```

File: app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua [hunk 3]
```diff
@@ -99,3 +99,31 @@ function SmoothnessFromSurface(surface)
...
+  if not scale then
+    return nil, nil, nil, "No MTB scale was given"
+  end
+  if Set({"0", "0+", "0-"})[scale] then
+    return "bad", "MTB scale to smoothness", "medium", nil
+  end 
+  return "very_bad", "MTB scale to smoothness", "medium", nil
+end
+
+function SmoothnessFromTrackType(type)
```
Comment (tordans) [inline app/process/roads_bikelanes/surfaceQuality/DeriveSmoothness.lua:113]:
```suggestion
-- Wiki https://wiki.openstreetmap.org/wiki/Key:tracktype
-- Categories that cyclosm.org use https://github.com/cyclosm/cyclosm-cartocss-style/wiki/Tag-to-Render
-- Mapping of Smoothness<>Surface https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery
-- Mapping of Smoothness<>Surface (Legacy) https://wiki.openstreetmap.org/wiki/Berlin/Verkehrswende/smoothness
function SmoothnessFromTrackType(type)
```

File: app/process/roads_bikelanes/surfaceQuality/SurfaceQuality.lua [hunk 4]
```diff
@@ -21,10 +21,16 @@ function SurfaceQuality(object)
   -- 1. `smoothess` tag
   -- 2. `smoothess` extrapolated from surface data
```
Comment (tordans) [inline app/process/roads_bikelanes/surfaceQuality/SurfaceQuality.lua:22]:
```suggestion
  -- 1. `smoothess` tag
  -- 2. `smoothess` extrapolated from `surface` data
  -- 3. `smoothess` extrapolated from `tracktype` tag, mostly on `highway=track`
  -- 4. `smoothess` extrapolated from `mtb:scale` tag, mostly on `highway=path`
```

**Review by tordans — CHANGES_REQUESTED** [review]
Some documentation and also small updated on the transformation after comparing streetcomplete's smoothness gallery.

**Pull request merged**